src/transformers/modeling_utils.py

```diff
@@ -2162,8 +2162,24 @@ def _get_resized_embeddings(
             dtype=old_embeddings.weight.dtype,
         )
 
-        # initialize all new embeddings (in particular added tokens)
-        self._init_weights(new_embeddings)
+        # initialize new embeddings (in particular added tokens) if `new_num_tokens` is larger
```

Comment:
one thing we need to make sure of is that this does not have issue with deepspeed and multinodes! (as for example the mean would require an all gather I think. This computation has to be done on gpu 0 is what my intuition tells me)

Reply:
That is absolutely right, thanks for mentioning that. I have added support for Deepspeed and tested it manually. I will add test cases for it tomorrow!

Reply:
@ArthurZucker I have added tests for deepspeed but they fail because the test env doesn't have deepspeed installed. what could be a solution for this? should I add the tests to `transformers/tests/deepspeed/test_deepspeed.py` for example?

Reply:
@abuelnasr0 when tests depend on a library like deepspeed, tag them with the `@require_deepspeed` decorator so the test runner knows how to handle them! You can search the codebase for other examples where it's used and just copy the imports/decorator from there.

Reply:
@Rocketknight1 Thank you for your response. It turns out that I was importing Deepspeed without checking if it was available. I fixed it.
@Rocketknight1 @ArthurZucker The code is ready for review now. Could you check it and see if the tests are all right?

Reply:
Sorry for the delay!

Reply:
@ArthurZucker  I have added 🔴 to the PR title. Is that what you meant? or should I add a `logger.warning()` to the code describing the new change in initializing embedding weights?

Reply:
>let's make sure we warn users about it!

I didn't see this comment the first time I opened the PR! I will add a warning to the user describing new changes.